BlockSettleUILib/Trading/DealerXBTSettlementContainer.cpp

```diff
@@ -286,7 +293,46 @@ void DealerXBTSettlementContainer::onTXSigned(unsigned int idReq, BinaryData sig
       logger_->debug("[DealerXBTSettlementContainer::onTXSigned] Payin sent");
 
       // ok. there is nothing this container could/should do
-      emit completed(id());
+//      emit completed(id());
+   }
+}
+
+void DealerXBTSettlementContainer::onZCReceived(const std::string &
+   , const std::vector<bs::TXEntry> &entries)
+{
+   const auto &cbTX = [this](const Tx &tx)
+   {
+      if (tx.getNumTxIn() != 1) {   // not a pay-out
+         return;
+      }
+      const auto &txIn = tx.getTxInCopy(0);
+      const auto &op = txIn.getOutPoint();
+      if (op.getTxHash() != expectedPayinHash_) {
+         return;  // not our pay-out
+      }
+      const auto &serializedTx = tx.serialize();
+      const auto &buyAuthKey = weSellXbt_ ? reqAuthKey_ : authKey_;
+      const auto &sellAuthKey = weSellXbt_ ? authKey_ : reqAuthKey_;
+      const auto &result = bs::TradesVerification::verifySignedPayout(serializedTx
+         , buyAuthKey.toHexStr(), sellAuthKey.toHexStr(), expectedPayinHash_
+         , bs::XBTAmount(amount_).GetValue(), utxoReservationManager_->feeRatePb()
+         , settlementIdHex_, settlAddr_.display());
+      if (result->success) {
```

Comment:
There is a very small chance that fee rate will be changed while we are waiting for TX on chain. I think that we could ignore it